From 059dc2351965cc4e6bccd3c9e57804960b6c0698 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 5 Oct 2026 01:22:12 +0800
Subject: [PATCH 2/2] AOSCOS: fix(evdev-mt-touchpad.c): enable natural
 scrolling by default

Per KDE 6 review session #2, this is handled on the libinput level.

Link: https://aosc.io/news/2026-07-05-minutes-kde-6.7-review-session-2#%E9%97%AE%E9%A2%98%E6%8E%92%E6%9F%A5
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 src/evdev-mt-touchpad.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/evdev-mt-touchpad.c b/src/evdev-mt-touchpad.c
index 81b803de..3b27f567 100644
--- a/src/evdev-mt-touchpad.c
+++ b/src/evdev-mt-touchpad.c
@@ -3227,9 +3227,8 @@ tp_scroll_config_natural_get_default(struct libinput_device *device)
 {
 	struct evdev_device *dev = evdev_device(device);
 
-	return (evdev_device_has_model_quirk(dev, QUIRK_MODEL_APPLE_TOUCHPAD) ||
-		evdev_device_has_model_quirk(dev,
-					     QUIRK_MODEL_APPLE_TOUCHPAD_ONEBUTTON));
+	/* AOSC OS: Enable natural scrolling by default. */
+	return 1;
 }
 
 static void
@@ -3239,7 +3238,6 @@ tp_init_scroll(struct tp_dispatch *tp, struct evdev_device *device)
 	tp_circular_scroll_init(tp, device);
 
 	evdev_init_natural_scroll(device);
-	/* Override natural scroll config for Apple touchpads */
 	device->scroll.config_natural.get_default_enabled =
 		tp_scroll_config_natural_get_default;
 	device->scroll.natural_scrolling_enabled =
-- 
2.56.0

